sway-features/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+                                Ok(())
+                            },
+                        )*
+                        "" => Ok(()),
+                        _ => Err(Error::UnknownFeature(feature.to_string())),
+                    }
+                }
+
+                pub fn enable_feature(&mut self, feature: Features, enabled: bool) {
+                    match feature {
+                        $(
+                            Features::[<$name:camel>] => {
+                                self.[<$name:snake>] = enabled
+                            },
+                        )*
+                    }
+                }
+
+                pub fn is_enabled_by_cfg(&self, cfg: &str) -> Result<bool, Error> {
+                    match cfg {
+                        $(
+                            stringify!([<experimental_ $name:snake>]) => Ok(self.[<$name:snake>]),
+                        )*
+                        _ => Err(Error::UnknownFeature(cfg.to_string()))
+                    }
+                }
+
+                $(
+                pub fn [<with_ $name:snake>](mut self, enabled: bool) -> Self {
+                    self.[<$name:snake>] = enabled;
+                    self
+                }
+                )*
+            }
+
+            // enum ExperimentalFeatureError {
+            //     $(
+            //         [<$name:camel Enabled>],
+            //         [<$name:camel Disabled>],
+            //     )*
+            // }
+
+            // impl std::fmt::Display for ExperimentalFeatureError {
+            //     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            //         let error = match self {
+            //             $(
+            //                 Self::[<$name:camel Enabled>] => stringify!(Feature [<$name:snake>] needs to be enabled),
+            //                 Self::[<$name:camel Disabled>] => stringify!(Feature [<$name:snake>] needs to be disabled),
+            //             )*
+            //         };
+            //         f.write_str(error)
+            //     }
+            // }
+        }
+    };
+}
+
+features! {
+    new_encoding = true,
+    "https://github.com/FuelLabs/sway/issues/5727",
+
+    storage_domains = false,
+    "https://github.com/FuelLabs/sway/pull/6466",
+}
+
+#[derive(Clone, Debug, Default, Parser)]
+pub struct CliFields {
+    /// Comma separated list of all experimental features that will be enabled
+    #[clap(long, value_delimiter = ',')]
+    pub experimental: Vec<Features>,
+
+    /// Comma separated list of all experimental features that will be disabled
+    #[clap(long, value_delimiter = ',')]
+    pub no_experimental: Vec<Features>,
+}
+
+#[derive(Debug)]
+pub enum Error {
+    ParseError(String),
+    UnknownFeature(String),
+}
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Error::ParseError(_) => f.write_str("ParserError"),
```

Comment:
Can we have some more user friendly errors here?
```suggestion
            Error::ParseError(feature) => f.write_fmt(format_args!("Experimental feature \"{feature}\" cannot be parsed.")),
```

Reply:
Done.